http-clients/src/main/java/com/palantir/remoting3/clients/UserAgents.java

```diff
@@ -73,7 +81,86 @@ private static String formatSingleAgent(UserAgent.Agent agent, Map<String, Strin
         return formatted.toString();
     }
 
-    static boolean isValidLibraryName(String serviceName) {
+
+    /**
+     * Parses the given string into a {@link UserAgent} or throws an {@link IllegalArgumentException} if the string does
+     * not conform to the user agent syntax restrictions.
+     */
+    public static UserAgent parse(String userAgent) {
+        return parseInternal(userAgent, false /* strict */);
+    }
+
+    /** Like {@link #parse}, but returns only the syntactically correct agent parts (, possibly zero). */
+    public static UserAgent tryParse(String userAgent) {
+        return parseInternal(userAgent, true /* lenient*/);
```

Comment:
nit, and we should baseline this, but there should be a space between the 't' and the asterisk

Reply:
https://github.com/palantir/gradle-baseline/issues/225

Reply:
fixed